### PR TITLE
fix(chips): implements missing adapter method notifyTrailingIconInteraction

### LINKED ISSUE #652: (Chips): Chip adapter is missing method `notifyTrailingIconInteraction`
Chip adapter is missing method `notifyTrailingIconInteraction`

also should include property `shouldRemoveOnTrailingIconClick` and the associated getter/setter's from the foundation.

### PR DESCRIPTION
**Pull request by mgr34**
Implements missing adapter method `notifyTrailingIconInteraction`. Along with introducing `shouldRemoveOnTrailingIconClick` as ChipProp. (mdc-web may consider renaming this to shouldRemoveOnTrailingIconInteraction?)

A successful merge would close #652 and #271. Additionally would cause a breaking change where `removeIcon` is renamed to `trailingIcon`. 

Screenshot tests were updated, however, golden.json may need attention.

---

I signed it

### CODE REVIEW

File: packages/chips/Chip.tsx [hunk 1]
```diff
@@ -26,19 +26,21 @@ import * as Ripple from '@material/react-ripple';
 import {MDCChipFoundation} from '@material/chips/dist/mdc.chips';
 
 export interface ChipProps extends Ripple.InjectedProps<HTMLDivElement> {
-  id?: string;
+  id: string;
```
Comment (moog16) [inline packages/chips/Chip.tsx]: This change is pending #649

File: test/unit/chips/Chip.test.tsx [hunk 2]
```diff
@@ -22,6 +30,36 @@ test('calls setSelected if props.selected is true (#foundation.setSelected)', ()
   assert.isTrue(wrapper.state().classList.has('mdc-chip--selected'));
 });
 
+
+test('renders a Chip with foundation.shouldRemoveOnTrailingIconClick set to true', () => {
+  const wrapper = shallow<Chip>(<Chip id='3' />);
+  assert.isTrue(wrapper.instance().foundation.getShouldRemoveOnTrailingIconClick());
+});
+
+test('#componentDidMount sets #foundaiton.shouldRemoveOnTrailingIconClick to false if prop false', () => {
```
Comment (moog16) [inline test/unit/chips/Chip.test.tsx]: spelling --> foundation

File: packages/chips/Chip.tsx (hunk 1 above)
Comment (mgr34) [reply packages/chips/Chip.tsx]: and had #649 been approved... ;)

I'll change this back.

**Comment by moog16** [comment]
@mgr34 looks like you have a conflict with readme.md

**Comment by mgr34** [comment]
@moog16 -- not just that it also looks like I accidentally committed my desire to update all dates to 2019...ugh. Hold on, this needs some work.

**Comment by mgr34** [comment]
@moog16 -- merge resolve and commit reverted. Thought I had stashed my attempt to update everything to 2019.

**Comment by moog16** [comment]
This is a breaking change...I will ammend the commit history. `removeIcon` --> `trailingIcon`

**Comment by mgr34** [comment]
to be clear I meant for this to be a breaking change and that removeIcon should now be trailingIcon. Semantically this is the logical choice.

**Comment by moog16** [comment]
yup that was my fault, thank you!

**Pull request merged**